server.js

```diff
@@ -7545,6 +7545,71 @@ camp.route(/^\/maven-metadata\/v\/(https?)\/(.+\.xml)\.(svg|png|gif|jpg|json)$/,
     });
 }));
 
+// badge-json url
+camp.route(/^\/json\/(https?.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(query, match, sendBadge, request) {
+  const urlEncoded = match[1];
+  const format = match[2];
+  const defaultLabel = 'badge json';
+  let badgeData = getBadgeData(defaultLabel, {});
+  try {
+    const url = encodeURI(decodeURIComponent(urlEncoded));
+    request(url, function dealWithData(err, res, buffer) {
+      if (checkErrorResponse(badgeData, err, res, 'url not found')) {
+        sendBadge(format, badgeData);
+        return;
+      }
+      try {
+        let data = JSON.parse(buffer);
```

Comment:
Could you move this transformation code into its own function in `lib/`, and make a list there of the supported parameters? It's great to have it in the web page, but feel like it ought to live with the code, too. By breaking this out into a separate (pure) function, it becomes very easy to test all this logic using data-driven tests. You'll see we've a bunch of these kind of tests now, using sazerac.